Mathlib/Topology/Order/IntermediateValue.lean

```diff
@@ -711,7 +710,6 @@ theorem Continuous.strictMonoOn_of_inj_rigidity {f : α → δ}
   replace : StrictMonoOn f (Icc x y) := StrictMonoOn.mono hf_mono_st this
   exact this (left_mem_Icc.mpr (le_of_lt hxy)) (right_mem_Icc.mpr (le_of_lt hxy)) hxy
 
-set_option backward.synthInstance.canonInstances false in -- See https://github.com/leanprover-community/mathlib4/issues/12532
```

Comment:
Do any references to #12532 remain?

Reply:
After this PR, there are 23 remaining references. A few of those are flagged by the linter - perhaps they could be removed also. (Would you prefer exhaustively removing these as flagged by the linter?)

Reply:
Update: with most occurrences of the linter addressed, there are 9 remaining references to this issue. I don't think *all* of them can be removed, but this PR shows quite a few can.